pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -615,52 +565,48 @@ func (r *ReconcileHyperConverged) aggregateComponentConditions(req *common.HcoRe
-
-			if _, conditionFound = req.Conditions[conditionsv1.ConditionUpgradeable]; !conditionFound { // (#chart 10)
-				req.Conditions.SetStatusCondition(conditionsv1.Condition{ // (#chart 11)
-					Type:    conditionsv1.ConditionUpgradeable,
-					Status:  corev1.ConditionFalse,
-					Reason:  commonProgressingReason,
-					Message: "HCO is not Upgradeable due to progressing components",
-				})
-			} // else - Upgradeable is already exists
-
-			if _, conditionFound = req.Conditions[conditionsv1.ConditionAvailable]; !conditionFound { // (#chart 12)
-				req.Conditions.SetStatusCondition(conditionsv1.Condition{ // (#chart 13)
-					Type:    conditionsv1.ConditionAvailable,
-					Status:  corev1.ConditionTrue,
-					Reason:  reconcileCompleted,
-					Message: reconcileCompletedMessage,
-				})
-			} // else - Available is already exists
+		if req.Conditions.HasCondition(conditionsv1.ConditionProgressing) { // (#chart 9)
+
+			req.Conditions.SetStatusConditionIfUnset(conditionsv1.Condition{ // (#chart 10,11)
+				Type:    conditionsv1.ConditionUpgradeable,
+				Status:  corev1.ConditionFalse,
+				Reason:  commonProgressingReason,
+				Message: "HCO is not Upgradeable due to progressing components",
+			})
+
+			req.Conditions.SetStatusConditionIfUnset(conditionsv1.Condition{ // (#chart 12,13)
+				Type:    conditionsv1.ConditionAvailable,
+				Status:  corev1.ConditionTrue,
+				Reason:  reconcileCompleted,
+				Message: reconcileCompletedMessage,
+			})
+
 		} else {
+
 			req.Conditions.SetStatusCondition(conditionsv1.Condition{ // (#chart 14)
 				Type:    conditionsv1.ConditionProgressing,
 				Status:  corev1.ConditionFalse,
 				Reason:  reconcileCompleted,
 				Message: reconcileCompletedMessage,
 			})
 
-			if _, conditionFound = req.Conditions[conditionsv1.ConditionUpgradeable]; !conditionFound { // (#chart 15)
-				req.Conditions.SetStatusCondition(conditionsv1.Condition{ // (#chart 16)
-					Type:    conditionsv1.ConditionUpgradeable,
-					Status:  corev1.ConditionTrue,
-					Reason:  reconcileCompleted,
-					Message: reconcileCompletedMessage,
-				})
-			}
+			req.Conditions.SetStatusConditionIfUnset(conditionsv1.Condition{ // (#chart 15,16)
+				Type:    conditionsv1.ConditionUpgradeable,
+				Status:  corev1.ConditionTrue,
+				Reason:  reconcileCompleted,
+				Message: reconcileCompletedMessage,
+			})
+
+			req.Conditions.SetStatusConditionIfUnset(conditionsv1.Condition{ // (#chart 17,18)
+				Type:    conditionsv1.ConditionAvailable,
+				Status:  corev1.ConditionTrue,
+				Reason:  reconcileCompleted,
+				Message: reconcileCompletedMessage,
+			})
 
-			if _, conditionFound = req.Conditions[conditionsv1.ConditionAvailable]; !conditionFound { // (#chart 17) {
-				req.Conditions.SetStatusCondition(conditionsv1.Condition{ // (#chart 18)
-					Type:    conditionsv1.ConditionAvailable,
-					Status:  corev1.ConditionTrue,
-					Reason:  reconcileCompleted,
-					Message: reconcileCompletedMessage,
-				})
-			}
 		}
```

Comment:
Worth a new function?

Reply:
Considered this (regarding these 3 comments) - I think it'll cause the overall flow to become less clear than is now.